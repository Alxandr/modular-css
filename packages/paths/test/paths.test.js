"use strict";

const dedent = require("dedent");
    
const Processor = require("modular-css-core");
const namer = require("test-utils/namer.js");
const { cwd, find } = require("test-utils/fixtures.js");

const paths = require("../paths.js");

describe("modular-css-paths", () => {
    it("should return a falsey value if a file isn't found", () => {
        const fn = paths({
            paths : [
                "./fixtures",
            ],
        });

        expect(fn(".", "./fooga.css")).toBeFalsy();
    });

    it("should return the absolute path if a file is found", () => {
        const fn = paths({
            paths : [
                find("one"),
            ],
        });

        expect(fn(".", "./one.css")).toBe(find("one.css"));
    });

    it("should check multiple paths for files & return the first match", () => {
        const fn = paths({
            paths : [
                find("one"),
                find("sub"),
            ],
        });

         expect(fn(".", "./sub.css")).toBe(find("sub.css"));
    });

    it("should be usable as a modular-css resolver", async () => {
        const processor = new Processor({
            cwd,
            namer,
            resolvers : [
                paths({
                    paths : [
                        find("sub"),
                        find("two"),
                    ],
                }),
            ],
        });
        
        await processor.string(
            "./one/start.css",
            dedent(`
                @value sub from "./sub.css";
                
                .rule {
                    composes: two from "./two.css";
                }
            `)
        );

        const result = await processor.output();
        
        expect(result.compositions).toMatchSnapshot();
    });
});
