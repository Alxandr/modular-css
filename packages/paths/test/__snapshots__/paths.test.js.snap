// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modular-css-paths should be usable as a modular-css resolver 1`] = `
Object {
  "one/start.css": Object {
    "rule": "two rule",
    "sub": "\\"sub\\"",
  },
  "one/sub/sub.css": Object {
    "sub": "sub",
  },
  "two/two.css": Object {
    "two": "two",
  },
}
`;
